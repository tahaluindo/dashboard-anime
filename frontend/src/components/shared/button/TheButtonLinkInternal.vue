<template>
  <router-link
    :to="to"
    class="duration-450 hover:duration-150"
    :class="activeClasses"
  >
    <slot />
  </router-link>
</template>

<script lang="ts">
import { computed, defineComponent, PropType } from "vue";
import { LocationAsRelativeRaw, useRoute } from "vue-router";
import { isMobile } from "../../../helpers/utils";

export default defineComponent({
  props: {
    to: {
      type: Object as PropType<LocationAsRelativeRaw>,
      required: true,
    },
    active: {
      type: String,
      default: "",
    },
    mobileActive: {
      type: String,
      default: "",
    },
  },
  setup(props) {
    const route = useRoute();

    const isActive = () => {
      return route.name === props.to.name;
    };

    const activeClasses = computed(() => {
      return [
        isActive() && props.active,
        isActive() && isMobile() && props.mobileActive,
      ];
    });

    return { activeClasses };
  },
});
</script>
