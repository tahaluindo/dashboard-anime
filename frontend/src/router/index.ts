import { createRouter, createWebHistory, RouteRecordRaw } from "vue-router";
import AddUser from "../views/AddUser.vue";
import AllUsers from "../views/AllUsers.vue";
import Home from "../views/Home.vue";
import User from "../views/User.vue";

const routes: Array<RouteRecordRaw> = [
  {
    path: "/",
    name: "Home",
    component: Home,
  },
  {
    path: "/usuarios",
    name: "AllUsers",
    component: AllUsers,
  },
  {
    path: "/usuario/:id",
    component: User,
    name: "User",
    props: true,
  },
  {
    path: "/adicionar-usuario",
    name: "AddUser",
    component: AddUser,
  },
];

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes,
  scrollBehavior() {
    return window.scrollTo({ top: 0, behavior: "smooth" });
  },
});

export default router;
