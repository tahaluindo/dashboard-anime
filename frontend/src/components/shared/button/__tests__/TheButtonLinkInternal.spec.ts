import { mount } from "@vue/test-utils";
import { LocationAsRelativeRaw } from "vue-router";
import router from "../../../../router/__mocks__/router.mock";
import { mockWindowVisualViewport } from "../../../../__mocks__/window.mock";
import TheButtonLinkInternal from "../TheButtonLinkInternal.vue";

describe("TheButtonLinkInternal component test", () => {
  it("should render the default component", async () => {
    mockWindowVisualViewport();

    router.push("/");
    await router.isReady();

    const to: LocationAsRelativeRaw = {
      name: "Home",
    };

    const active = "teste";
    const mobileActive = "teste";

    const wrapper = mount(TheButtonLinkInternal, {
      props: {
        to,
        active,
        mobileActive,
      },
      global: {
        plugins: [router],
      },
    });

    const result = wrapper.vm.activeClasses;

    expect(result.toString()).toBe("teste,teste");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
