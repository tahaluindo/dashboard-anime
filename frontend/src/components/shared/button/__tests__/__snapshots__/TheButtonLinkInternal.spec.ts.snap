// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheButtonLinkInternal component test should render the default component 1`] = `"<a aria-current=\\"page\\" href=\\"/\\" class=\\"router-link-active router-link-exact-active duration-450 hover:duration-150 teste teste\\"></a>"`;
